core/include/mm/vm.h

```diff
@@ -116,12 +116,25 @@ TEE_Result vm_va2pa(const struct user_mode_ctx *uctx, void *ua, paddr_t *pa);
  */
 TEE_Result vm_pa2va(const struct user_mode_ctx *uctx, paddr_t pa, void **va);
 
+/*
+ * vm_check_access_rights - Check access permission against a memory buffer
+ *
+ * @uctx: User mode context
+ * @flags: Permission flags to check against
+ * @uaddr: Base address of the user buffer to test
+ * @len: Byte size of the user buffer to test
+ *
+ * Return TEE_SUCCESS or TEE_ERROR_ACCESS_DENIED if buffer location is valid,
+ * else return another TEE_Result code.
+ */
 TEE_Result vm_check_access_rights(const struct user_mode_ctx *uctx,
 				  uint32_t flags, uaddr_t uaddr, size_t len);
 
-/*-----------------------------------------------------------------------------
- * If ctx is NULL user mapping is removed and ASID set to 0
- *---------------------------------------------------------------------------*/
+/*
+ * vm_set_ctx - Set memory mapping context
+ *
+ * @ctx: Session context, if NULL user mapping is removed and ASID set to 0
```

Comment:
Mentioning ASID in a non-arch-specific file?

Reply:
True. No need to mention ASID. User mapping is ok?

Reply:
Yes